unstructured/ingest/processor.py

```diff
@@ -74,15 +87,28 @@ def run(self):
             if not docs:
                 return
 
+        # get a create_session_handle function if the connector supports it
+        create_session_handle_fn = (
+            partial(
+                cast(ConnectorSessionHandleMixin, self.doc_connector).create_session_handle,
+                cast(BaseConnector, self.doc_connector).config,
+            )
+            if isinstance(self.doc_connector, ConnectorSessionHandleMixin)
+            else None
```

Comment:
i don't think this handle should actually exist in the parent process. but i think i see the issue that the init process needs connection info, at the very least, following this approach.

i'm starting to think the cleanest way to do this is for the subprocess itself to create the SessionHandle lazily for the first IngestDoc it processes, since it will have the connector config at that time.

Reply:
or, it could pass the connector config of the first IngestDoc in `initargs` for the sub process's `initializer`, and then connector can choose to create its own SessionHandle if applicable.

Reply:
coming back to this fresh, I think I was over-engineering in case we ever needed to handle subprocess work async, but starting to feel like it's safe to just assume this is always serial within that subprocess. I like the idea of just lazily creating it, will head with that.

also, I'm now thinking the config should just own the definition for how to create the session handle. this would also be cleaner if we move that logic inside the subprocess since we don't pass the connector itself through (I don't believe).